### PR TITLE
Mkdocs setup

### LINKED ISSUE #1343: Responsive and friendly Documentation page
## Checklist

* [X] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [X] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
* [x] Did you perform a search about this feature? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?

A clear and concise description of what problem this feature would solve. For example:

- It makes it easy for users to consume the guides about this open source project

## Solution description

A detailed description of what you want to happen.

Having a docs.domain.com or domain.com/docs page is really great and gets very fast accessibility to users.

### LINKED ISSUE #1380: Drop markdown-toc in favor of default GitHub toc
## Checklist

* [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/main/docs/v2/manual.md).
* [x] Did you perform a search about this feature? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?

- the `./docs/v*/manual.md` files use the `markdown-toc` tool to generate an embedded table of contents
- the decision to use this tool [predates (by many years) the existence of default GitHub table of contents support](https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/)
- use of `markdown-toc` requires a reasonably modern `node` installation and out-of-band installation steps

## Solution description

Drop the `markdown-toc` usage in favor of GitHub's built-in default table of contents support.

## Describe alternatives you've considered

N/A (?)

### PR DESCRIPTION
**Pull request by meatballhat**
## What type of PR is this?

- cleanup
- documentation

## What this PR does / why we need it:

Adds initial mkdocs material configuration and index doc so that a docs site may be hosted via GitHub Pages.

## Which issue(s) this PR fixes:

Fixes #1343 
Fixes #1380 

## Special notes for your reviewer:

There was some mention in #1343 about possibly using docusaurus, but I threw this together given that other maintainers are either already familiar with mkdocs or prefer it to docusaurus.

### CODE REVIEW

File: mkdocs-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+mkdocs
```
Comment (rliebz) [inline mkdocs-requirements.txt:1]: Do we want to pin versions here?

**Review by rliebz — DISMISSED** [review]

File: mkdocs-requirements.txt (hunk 1 above)
Comment (meatballhat) [reply mkdocs-requirements.txt:1]: Yes please 👍🏼 I'm assuming there will be other missing bits needed to make the whole thing work correctly. WDYT about me adding these `make` targets to this PR along with a blurb in `docs/CONTRIBUTING.md`?

- `make docs-deps`
- `make docs`
- `make serve-docs`

Comment (rliebz) [reply mkdocs-requirements.txt:1]: Makes sense to me!

**Comment by meatballhat** [comment]
FYI out of band I've enabled GitHub Pages and added a CNAME record of:

```
cli.urfave.org. 3600 IN CNAME urfave.github.io.
```

**Comment by saintmalik** [comment]
> FYI out of band I've enabled GitHub Pages and added a CNAME record of:
> 
> ```
> cli.urfave.org. 3600 IN CNAME urfave.github.io.
> ```

Great, nice, sorry it took me some days to respond, can i help out also?

**Comment by saintmalik** [comment]
Also noticed something: the pen editing button leads to a 404 page at the moment.

https://github.com/urvafe/cli/edit/main/docs/v2/index.md

**Comment by meatballhat** [comment]
@saintmalik What do you think of this PR? Reasonable enough as a first attempt?

**Comment by saintmalik** [comment]
Yeah, its pretty awesome, and makes life easy for everyone

**Comment by kolyshkin** [comment]
Hint: if you use `git pull --rebase` you can get rid of merge commits.

**Comment by meatballhat** [comment]
> Hint: if you use `git pull --rebase` you can get rid of merge commits.

@kolyshkin yep I'm aware 😁 I realize the whole merge/rebase debate has a long history that I definitely don't want to engage in here. Suffice to say I'm in favor of adapting my preferences to what works for fellow contributors. Any chance you're up for opening a [discussion](https://github.com/urfave/cli/discussions) if you'd like to see a rebase-centric set of workflows and norms?

**Pull request merged**